### DISCUSSION TITLE
Error when Running on devnet

### DISCUSSION THREAD

**Discussion by RyanKoech** [reach-sh/reach-lang#878] [Q&A] [answered]
I get this error when I run ```REACH_CONNECTOR_MODE=ALGO ./reach devnet ```
`Wait for reach-devnet-algo-pg:5432
Not starting debugger. To start, use REACH_DEBUG=1.
Starting algod
Checking for algod.net
Starting indexer
Waiting for logs...
Logging to:  /var/lib/algorand_network/Primary/node.log`
The log in the terminal is infinite. Please help.

**Comment by jeapostrophe** [comment, marked as answer]
What makes you think this is an error?

**Reply by RyanKoech** [reply]
Because it logs endlessly, and the front end does not respond.

**Reply by jeapostrophe** [reply]
It should log endlessly, because it is continuously running doing work.

You didn't mention that you were running a front-end at all. How are you running it? What is it? What is it doing?

**Reply by RyanKoech** [reply]
The  button to connect to the wallet is not responsive. Plus, the logs seems to mentioning errors so i assumed so.
```

INDEX:{"error":"Get \"http://[::]:4180/genesis\": dial tcp [::]:4180: connect: cannot assign requested address","level":"error","msg":"unable to fetch genesis from algod","time":"2022-04-29T15:06:37Z"}
INDEX:Indexer died, restarting...
INDEX:serving on :8980
INDEX:{"level":"info","msg":"serving on :8980","time":"2022-04-29T15:06:37Z"}
INDEX:{"level":"info","msg":"Running 0 migrations.","time":"2022-04-29T15:06:37Z"}
INDEX:{"level":"info","msg":"Setting status: Migrations Complete","time":"2022-04-29T15:06:37Z"}
INDEX:{"level":"info","msg":"Migration finished successfully.","time":"2022-04-29T15:06:37Z"}
INDEX:{"level":"info","msg":"fetching genesis from algod","time":"2022-04-29T15:06:37Z"}
INDEX:{"error":"Get \"http://[::]:4180/genesis\": dial tcp [::]:4180: connect: cannot assign requested address","level":"error","msg":"unable to fetch genesis from algod","time":"2022-04-29T15:06:37Z"}

```

**Reply by jeapostrophe** [reply]
Those messages are not suspicious to me. I'd be looking at what your browser console log is saying when you try to connect to the wallet. I assume that you've run the devnet multiple times, correct?

**Reply by jeapostrophe** [reply]
(After running `reach down`)

**Reply by RyanKoech** [reply]
Nope. It was my first time trying.
I have looked at the terminal, and I've seen this error thrown when I click the connect wallet button.

``` caught (in promise) TypeError: Reach.setSignStrategy is not a function ```

I traced the error to this line of code `Reach.setSignStrategy('AlgoSigner');`  of which when I comment it out it throws the following error:

``` Error: Providers created by environment do not have default addresses. Calling setProviderByEnv or setProviderByName removes this capability. Try removing calls to those functions. ```

Which I have traced to the preceding line of code. ` Reach.setProviderByName('TestNet'); `

ConnectWallet.js looks as below

```
    const connectWallet = async () => {        
        Reach.setSignStrategy('AlgoSigner');
        Reach.setProviderByName('TestNet');
        const acc = await Reach.getDefaultAccount();
        const balAtomic = await Reach.balanceOf(acc);
        const bal = Reach.formatCurrency(balAtomic, 4);
        setAccount(Object.assign({}, acc));
        setBalance(bal);
    }
```

**Reply by jeapostrophe** [reply]
The error tells you exactly what the problem is.

1) There is no function named `setSignStrategy` --- https://docs.reach.sh/frontend/ --- and this function does not occur in any documentation or examples.

2) You are calling `getDefaultAccount` after calling `setProviderByName`, which cannot work.

**Reply by RyanKoech** [reply]
Thanks, I'm sorry for the hustle, but I am still learning reach and therefore struggling to get the ConnectWallet.js working to provide some test accounts. Ideally, how would you set it up? Test accounts, which I can fund from the Faucet. Thanks!

**Reply by RyanKoech** [reply]
I have tried this, which I don't think it's fine anyway:
```

    const startingBalance = Reach.parseCurrency(100);

    const connectWallet = async () => {      
        const acc = await Reach.newTestAccount(startingBalance);
        const balAtomic = await Reach.balanceOf(acc);
        const bal = Reach.formatCurrency(balAtomic, 4);
        setAccount(Object.assign({}, acc));
        setBalance(bal);
    }
```
I get an overspend error: `Object { type: "signAndPost", e: Error, es: "Error: Network request error. Received status 400: TransactionPool.Remember: transaction UR7EKQOMQDGUBUKB7FGPTDWBC5WB5GXZ3WQGHOVHY2A3KFKEHP5Q: overspend (account LGMPWNMXPSKTEA3NF4WJAPIBG4QCDE7NME3MMKBKN7VNSQENXIRIHBBTGE`

**Reply by RyanKoech** [reply]
> Nope. It was my first time trying. I have looked at the terminal, and I've seen this error thrown when I click the connect wallet button.
> 
> `caught (in promise) TypeError: Reach.setSignStrategy is not a function`
> 
> I traced the error to this line of code `Reach.setSignStrategy('AlgoSigner');` of which when I comment it out it throws the following error:
> 
> `Error: Providers created by environment do not have default addresses. Calling setProviderByEnv or setProviderByName removes this capability. Try removing calls to those functions.`
> 
> Which I have traced to the preceding line of code. `Reach.setProviderByName('TestNet');`
> 
> ConnectWallet.js looks as below
> 
> ```
>     const connectWallet = async () => {        
>         Reach.setSignStrategy('AlgoSigner');
>         Reach.setProviderByName('TestNet');
>         const acc = await Reach.getDefaultAccount();
>         const balAtomic = await Reach.balanceOf(acc);
>         const bal = Reach.formatCurrency(balAtomic, 4);
>         setAccount(Object.assign({}, acc));
>         setBalance(bal);
>     }
> ```

I just wanted to point out, the coding approach above in ConnectWallet.js is one that I have borrowed from many other Reach projects that I have seen implement it from the Reach YouTube channel. What has changed?

**Reply by jeapostrophe** [reply]
My hypothesis is that you are attached to the "real" network (TestNet or MainNet) because you can't use `newTestAccount` for those, but need to use a faucet like https://bank.testnet.algorand.network/

**Reply by RyanKoech** [reply]
How do I use the local network to be able to use the private faucet? Because the code below does work for my CLI .mjs code.

```
    const startingBalance = Reach.parseCurrency(100);
        const acc = await Reach.newTestAccount(startingBalance);
        const balAtomic = await Reach.balanceOf(acc);
```

**Reply by RyanKoech** [reply]
In another project, a react project (```./reach react ```), I have been able to do that.

```
  
  async componentDidMount() {
    reach.setProviderByName("LocalHost")
    const acc = await reach.newTestAccount(startingBalance);
    const balAtomic = await reach.balanceOf(acc);
    const bal = reach.formatCurrency(balAtomic, 4);
    this.setState({acc, bal});
    if (await reach.canFundFromFaucet()) {
      this.setState({view: 'FundAccount'});
    } else {
      this.setState({view: 'DeployerOrAttacher'});
    }
  }
```

Now I want to be able to achieve this, on the previous one ```./reach devnet``` because I am not developing with react.

**Reply by jeapostrophe** [reply]
Do the same thing as when you were using `reach react`

**Reply by RyanKoech** [reply]
> 

I have and the error is still the same
`Object { type: "signAndPost", e: Error, es: "Error: Network request error. Received status 400: TransactionPool.Remember: transaction BZILSTQFVI2MUFK7OJG726TCIHH3COC4NIOJNQF7YCTNTDTA7YAQ: overspend (account LGMPWNMXPSKTEA3NF4WJAPIBG4QCDE7NME3MMKBKN7VNSQENXIRIHBBTGE, data {AccountBaseData:{Status:Offline MicroAlgos:{Raw:0} RewardsBase:0 RewardedMicroAlgos:{Raw:0} AuthAddr:AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAY5HFKQ`

**Reply by jeapostrophe** [reply]
What version of the stdlib and devnet are you using? `reach hashes` tells you the second and the `stdlib` is determined by your `package.json`

**Reply by RyanKoech** [reply]
When I run ``` reach hashes ```
```
reach-cli: 9f925e05
reach: 9f925e05
react-runner: 9f925e05
rpc-server: 9f925e05
runner: 9f925e05
devnet-algo: 9f925e05
```
For the ```stdlib``` its version ```0.1.9```

**Reply by jeapostrophe** [reply]
This means your devnet is 0.1.10 but your stdlib is 0.1.9, so they mismatch and the faucet is empty.

**Reply by DanBurton** [reply]
You can likely update your stdlib to the latest by running `npm install @reach-sh/stdlib@latest`
